test/common/stats/thread_local_store_test.cc

```diff
@@ -848,6 +848,27 @@ TEST_F(StatsThreadLocalStoreTest, ExtractAndAppendTagsRegexValueWithMatch) {
   EXPECT_EQ("bar", symbol_table_.toString(tags[1].second));
 }
 
+TEST_F(StatsThreadLocalStoreTest, ExtractAndAppendTagsRegexBuiltinExpression) {
+  store_->initializeThreading(main_thread_dispatcher_, tls_);
+
+  envoy::config::metrics::v3::StatsConfig stats_config;
+  store_->setTagProducer(std::make_unique<TagProducerImpl>(stats_config));
+
+  StatNamePool pool(symbol_table_);
+  StatName name = pool.add("cluster.foo.bar");
+  StatName tag_key = pool.add("a");
+  StatName tag_val = pool.add("b");
+
+  StatNameTagVector tags{{tag_key, tag_val}};
+  store_->extractAndAppendTags(name, pool, tags);
+
+  EXPECT_EQ(2, tags.size());
```

Comment:
nit: this should be ASSERT_EQ -- that's the pattern we use for a check that, if it fails, the test-method should exit right after that to avoid crashing and thus not running the rest of the tests.

The others can be EXPECT_EQ.